core/src/main/java/org/infinispan/cache/impl/CacheImpl.java

```diff
@@ -285,17 +286,15 @@ private V compute(K key, BiFunction<? super K, ? super V, ? extends V> remapping
    }
 
    private V computeInternal(K key, BiFunction<? super K, ? super V, ? extends V> remappingFunction, boolean computeIfPresent, Metadata metadata, long flags) {
-      return computeInternal(key, remappingFunction, computeIfPresent, metadata, flags, getInvocationContextWithImplicitTransaction(false, 1));
+      return computeInternal(key, remappingFunction, computeIfPresent, metadata, flags, this::singleKeyContextWithImplicitTransaction);
```

Comment:
Please cache the method reference - this is a capturing method handle => allocation.

Reply:
I'm not a big fan of caching this kind of stuff. It is memory always used, independently if the cache is used or not. and the cache already has a lot of fields. 

only if you insist, I'll change it :)

Reply:
I dislike caching method references as well (ideally this should be done by our clever JVM), but when the con is an allocation on each method call, I think that such ugly optimization makes sense. @danberindei wdyt?

Reply:
@pruivo, I agree with Radim: if storing one lambda in a `CacheImpl` field is too much, then creating a new lambda instance on every invocation is much much worse.

I see in https://github.com/infinispan/infinispan/pull/5458 I only added 2 fields, but I have no idea now why I closed that PR.